cmd/mesh/manifest-versions_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mesh
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	goversion "github.com/hashicorp/go-version"
+
+	"istio.io/operator/pkg/version"
+)
+
+func TestGetVersionCompatibleMap(t *testing.T) {
+	type args struct {
+		versionsURI string
+		binVersion  *goversion.Version
+		l           *logger
+	}
+	goVer131000, _ := goversion.NewVersion("1.3.1000")
+	goVer132, _ := goversion.NewVersion("1.3.2")
+	scVer132, _ := goversion.NewConstraint(">=1.3.0,<=1.3.2")
+	rcVer132, _ := goversion.NewConstraint("1.3.2")
+	vm132 := &version.CompatibilityMapping{
+		OperatorVersion:          goVer132,
+		SupportedIstioVersions:   scVer132,
+		RecommendedIstioVersions: rcVer132,
+	}
+	l := newLogger(true, os.Stdout, os.Stderr)
+	tests := []struct {
+		name    string
+		args    args
+		want    *version.CompatibilityMapping
+		wantErr error
+	}{
+		{
+			name: "built-in version map",
+			args: args{
+				versionsURI: "__nonexistent-versions.yaml",
+				binVersion:  goVer132,
+				l:           l,
+			},
+			want:    vm132,
+			wantErr: nil,
+		},
+		{
+			name: "read from github",
```

Comment:
I suggest skipping the read from github unless you feel like implementing a server in the test so that it's hermetic. We don't want to be calling out to a real URL in a test.
Instead, I suggest pointing it to a file in testdata. This will allow you also to test the negative case if path is incorrect, should get an error.

Reply:
Sure, will use test data instead.